include/boost/unordered/detail/map.hpp

```diff
@@ -18,6 +19,7 @@ namespace boost {
         typedef std::pair<K const, M> value_type;
         typedef H hasher;
         typedef P key_equal;
+        typedef K key_type;
         typedef K const const_key_type;
```

Comment:
Why has you added this `typedef`? Seems obvious it was missing, but is there something that used to fail without it?

Reply:
Good catch, I used it in an earlier draft, but I forgot to remove it later. I'll remove it here, it's not needed